example-mlem-get-started/generate.sh

```diff
@@ -0,0 +1,156 @@
+#!/bin/bash
+# See https://dvc.org/doc/tutorials/get-started
+
+# Setup script env:
+#   e   Exit immediately if a command exits with a non-zero exit status.
+#   u   Treat unset variables as an error when substituting.
+#   x   Print commands and their arguments as they are executed.
+set -eux
+
+HERE="$( cd "$(dirname "$0")" ; pwd -P )"
+REPO_NAME="example-mlem-get-started"
+
+BUILD_PATH="$HERE/build"
+
+pushd $BUILD_PATH
+if [ ! -d "$BUILD_PATH/.venv" ]; then
+  exit 1
+  virtualenv -p python3 .venv
+  export VIRTUAL_ENV_DISABLE_PROMPT=true
+  source .venv/bin/activate
+  echo '.venv/' > .gitignore
+  pip install gitpython
+  pip install "git+https://github.com/iterative/mlem#egg=mlem[all]" --use-deprecated=legacy-resolver
+  pip install -r $HERE/code/src/requirements.txt
+fi
+popd
+
+source $BUILD_PATH/.venv/bin/activate
+
+REPO_PATH="$HERE/build/$REPO_NAME"
+
+if [ -d "$REPO_PATH" ]; then
+  echo "Repo $REPO_PATH already exists, please remove it first."
+  exit 1
+fi
+
+TOTAL_TAGS=15
+STEP_TIME=100000
+BEGIN_TIME=$(( $(date +%s) - ( ${TOTAL_TAGS} * ${STEP_TIME}) ))
+export TAG_TIME=${BEGIN_TIME}
+export GIT_AUTHOR_DATE=${TAG_TIME}
+export GIT_COMMITTER_DATE=${TAG_TIME}
+tick(){
+  export TAG_TIME=$(( ${TAG_TIME} + ${STEP_TIME} ))
+  export GIT_AUTHOR_DATE=${TAG_TIME}
+  export GIT_COMMITTER_DATE=${TAG_TIME}
+}
+
+export GIT_AUTHOR_NAME="Mikhail Sveshnikov"
+export GIT_AUTHOR_EMAIL="mike0sv@gmail.com"
+export GIT_COMMITTER_NAME="$GIT_AUTHOR_NAME"
+export GIT_COMMITTER_EMAIL="$GIT_AUTHOR_EMAIL"
+
+
+
+mkdir -p $REPO_PATH
+pushd $REPO_PATH
+
+git init -b main
+cp $HERE/code/README.md .
+cp $HERE/.gitattributes .
+git add .
+tick
+git commit -m "Initialize Git repository"
+git tag -a "0-git-init" -m "Git initialized."
+
+mlem init
+tick
+git add .mlem/config.yaml
+git commit -m "Initialize MLEM project"
+git tag -a "1-mlem-init" -m "MLEM initialized."
+
+
+cp $HERE/code/src/requirements.txt .
+cp $HERE/code/src/prepare.py .
+git add .
+python prepare.py
+git add .mlem/dataset
+tick
+git commit -m "Create data stage"
+git tag -a "2-prepare-stage" -m "Data created."
+
+
+cp $HERE/code/src/train.py .
+python train.py
+git add .mlem/model train.py
+tick
+git commit -m "Create train stage"
+git tag -a "3-train-stage" -m "Model trained."
+
+
+cp $HERE/code/src/evaluate.py .
+python evaluate.py
+git add metrics.json evaluate.py
+tick
+git commit -m "Evaluate model"
+git tag -a "4-eval-stage" -m "Metrics calculated"
+
+git rm -r --cached .mlem/model/rf .mlem/dataset/train.csv .mlem/dataset/test_x.csv .mlem/dataset/test_y.csv
+git commit -m "stop tracking data"
+
+dvc init
+dvc remote add myremote azure://example-mlem --default
+mlem config set default_storage.type dvc
+python train.py
+dvc add .mlem/model/rf .mlem/dataset/train.csv .mlem/dataset/test_x.csv .mlem/dataset/test_y.csv
```

Comment:
that's not very common for DVC to track each independently (esp dataset) - is it intended?

Reply:
there are .mlem files in this dir. Can we do something like `echo *.mlem > .dvcignore && dvc add data`?

Reply:
the correct pattern was found and it is `/**/?*.mlem`. Btw, does gitignore support ! negation?

Reply:
this does not seem to work, since we cant `dvc add .mlem/dataset/` since we lose `.mlem` files, even though they ignored. So going with `dvc add .mlem/dataset/*.csv`. Would be great to have some way to do "add everything except dvc-ignored"

Reply:
> Btw, does dvcignore support ! negation?

yes, it does support it

it should be working, otherwise it's a bug to my mind

Reply:
```
$ cat .dvcignore 
/**/*.mlem
!/.mlem/
(.venv) (mlem) MacBook-Pro-5:example-mlem-get-started mike0sv$ dvc check-ignore .mlem -d
.dvcignore:2:!/.mlem/   .mlem
$ cat .dvcignore 
/**/*.mlem
!/.mlem/
$ dvc check-ignore .mlem -d
.dvcignore:2:!/.mlem/   .mlem
```
I was trying to make an exception for `.mlem` dir in a root, but this is what I've got

Reply:
q - why do you need an exception like this? (curious)

does the same combination work for gitignore?

Reply:
My goal is to exclude all `*.mlem` files, but not `.mlem` dir in repo root. But simple `/**/*.mlem` adds `.mlem` dir to ignore list. With git it works